src/xia2/Wrappers/Dials/Functional/Merge.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+
+import libtbx.phil
+from dials.array_family import flex
+from dials.command_line import merge
+from dials.command_line.merge import phil_scope
+from dxtbx.model import ExperimentList
+
+from xia2.Driver.timing import record_step
+from xia2.lib.bits import _get_number
+from xia2.Modules.SSX.util import log_to_file, run_in_directory
+from xia2.Wrappers.Dials.Functional import diff_phil_from_params_and_scope, handle_fail
+
+xia2_logger = logging.getLogger(__name__)
+
+
+class Merge:
+    def __init__(self, working_directory: Path | None = None) -> None:
+        if working_directory:
+            self._working_directory = working_directory
+        else:
+            self._working_directory = Path.cwd()
+
+        self.params: libtbx.phil.scope_extract = phil_scope.extract()
+        self.working_directory = working_directory
+        self.output_filename = "merged.mtz"
+
+    def set_d_min(self, d_min: float) -> None:
+        self.params.d_min = d_min
+
+    def set_wavelength_tolerance(self, wavelength_tolerance: float) -> None:
+        self.params.wavelength_tolerance = wavelength_tolerance
+
+    def set_r_free_params(self, r_free_params: libtbx.phil.scope_extract) -> None:
+        self.params.r_free_flags = r_free_params
+
+    def set_assess_space_group(self, assess_space_group: bool) -> None:
+        self.params.assess_space_group = assess_space_group
+
+    def set_output_filename(self, filename: str) -> None:
```

Comment:
My idea for the new-style wrappers would be to use python properties rather than get and set methods. This is not applicable to setting properties on params (I think), but for other class members like output_filename we would have it as a private member (`self._output_filename`) and define it as a `@property` and define a method with a setter decorator.

Reply:
Although that could look a bit inconsistent if we have lots of "set_" methods for things that are set on params and then properties for the rest 🤔 .

Reply:
I agree with properties in general but I suspect that these getters/setters are a very heavily embedded idiom...